tiledb/sm/filesystem/posix.h

```diff
@@ -35,23 +35,22 @@
 
 #ifndef _WIN32
 
-#include <ftw.h>
-#include <sys/types.h>
-
 #include <functional>
 #include <string>
 #include <vector>
 
-#include "tiledb/sm/buffer/buffer.h"
 #include "tiledb/sm/config/config.h"
 #include "tiledb/sm/filesystem/filelock.h"
 #include "tiledb/sm/misc/status.h"
-#include "tiledb/sm/misc/thread_pool.h"
-#include "tiledb/sm/misc/uri.h"
+
+struct FTW;
```

Comment:
don't forward declare identifiers from system headers

Reply:
You're right, I missed that part, sorry. I was thinking that it was an external header file but even it is an external structure, it should not be forwarded.